tests/test_connections/test_ledger/test_ledger_api.py

```diff
@@ -363,7 +363,7 @@ async def test_send_signed_transaction_ethereum(
             response_message.transaction_digest.ledger_id
             == request.signed_transaction.ledger_id
         )
-
+        await asyncio.sleep(0.01)
```

Comment:
this resolved the non-deterministic failure in `tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_send_signed_transaction_ethereum`

Reply:
Could you explain why this solves the issue? Why do we need to wait here?

Reply:
I thought about it, and I'm not quite sure why this is now needed whereas it was not before. I can explain my reasoning to some extend: the failures were non-deterministic and, since the async is the only source of randomness, some "instinct" perhaps, for lack of a better word, led me to add a bit of sleep here. I realize this answer is not satisfactory but it's all I have :(

Reply:
I think this is the reason:

https://github.com/valory-xyz/open-autonomy/pull/1025#discussion_r916499210